doxygen/man/man3/LiveBar.3
.TH "LiveBar" 3 "Pn, 11 mar 2013" "Version 0.0.3" "AstroRush" \" -*- nroff -*-
.ad l
.nh
.SH NAME
LiveBar \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <LiveBar\&.hpp>\fP
.PP
Dziedziczy \fBStandardReferences\fP\&.
.SS "Metody publiczne"

.in +1c
.ti -1c
.RI "\fBLiveBar\fP ()"
.br
.ti -1c
.RI "void \fBreset\fP ()"
.br
.ti -1c
.RI "void \fBdraw\fP ()"
.br
.ti -1c
.RI "void \fBupdate\fP (const float &dt)"
.br
.ti -1c
.RI "void \fBaddBonus\fP ()"
.br
.ti -1c
.RI "bool \fBuseBonus\fP ()"
.br
.in -1c
.SS "Statyczne metody publiczne"

.in +1c
.ti -1c
.RI "static Uint32 \fBdisableImmortal_callbackTimer\fP (Uint32 interval, void *param)"
.br
.ti -1c
.RI "static bool \fBisALive\fP ()"
.br
.ti -1c
.RI "static void \fBcolision\fP (short type)"
.br
.ti -1c
.RI "static void \fBincreaseLive\fP ()"
.br
.ti -1c
.RI "static string \fBgetResult\fP ()"
.br
.ti -1c
.RI "static ulong \fBgetResultNum\fP ()"
.br
.ti -1c
.RI "static float \fBgetLiveAmount\fP ()"
.br
.ti -1c
.RI "static float \fBgetMapSpeed\fP ()"
.br
.ti -1c
.RI "static void \fBnextMeter\fP ()"
.br
.ti -1c
.RI "static void \fBresetMeter\fP ()"
.br
.in -1c
.SS "Dodatkowe Dziedziczone Składowe"
.SH "Opis szczegółowy"
.PP 
Definicja w linii 10 pliku LiveBar\&.hpp\&.
.SH "Dokumentacja konstruktora i destruktora"
.PP 
.SS "LiveBar::LiveBar ()"

.PP
Definicja w linii 10 pliku LiveBar\&.cpp\&.
.SH "Dokumentacja funkcji składowych"
.PP 
.SS "void LiveBar::addBonus ()\fC [inline]\fP"

.PP
Definicja w linii 20 pliku LiveBar\&.hpp\&.
.SS "void LiveBar::colision (shorttype)\fC [static]\fP"

.PP
Definicja w linii 81 pliku LiveBar\&.cpp\&.
.SS "Uint32 LiveBar::disableImmortal_callbackTimer (Uint32interval, void *param)\fC [static]\fP"
Metoda statyczna wywolywana przez timer wtedy kiedy mija czas w ktorym gracz nie traci zycia 
.PP
Definicja w linii 111 pliku LiveBar\&.cpp\&.
.SS "void LiveBar::draw ()"
Rysowanie paska zycia\&.\&.\&.cala gorna czesc ekranu 
.PP
Definicja w linii 20 pliku LiveBar\&.cpp\&.
.SS "static float LiveBar::getLiveAmount ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 48 pliku LiveBar\&.hpp\&.
.SS "static float LiveBar::getMapSpeed ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 49 pliku LiveBar\&.hpp\&.
.SS "static string LiveBar::getResult ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 45 pliku LiveBar\&.hpp\&.
.SS "static ulong LiveBar::getResultNum ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 46 pliku LiveBar\&.hpp\&.
.SS "void LiveBar::increaseLive ()\fC [static]\fP"

.PP
Definicja w linii 117 pliku LiveBar\&.cpp\&.
.SS "static bool LiveBar::isALive ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 30 pliku LiveBar\&.hpp\&.
.SS "static void LiveBar::nextMeter ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 51 pliku LiveBar\&.hpp\&.
.SS "void LiveBar::reset ()"

.PP
Definicja w linii 123 pliku LiveBar\&.cpp\&.
.SS "static void LiveBar::resetMeter ()\fC [inline]\fP, \fC [static]\fP"

.PP
Definicja w linii 52 pliku LiveBar\&.hpp\&.
.SS "void LiveBar::update (const float &dt)"

.PP
Definicja w linii 62 pliku LiveBar\&.cpp\&.
.SS "bool LiveBar::useBonus ()"

.PP
Definicja w linii 94 pliku LiveBar\&.cpp\&.

.SH "Autor"
.PP 
Wygenerowano automatycznie z kodu źródłowego programem Doxygen dla AstroRush\&.


src/LiveBar.cpp
#include "LiveBar.hpp"
#include "App.hpp"

ulong LiveBar::pDistNum;
float LiveBar::pLiveAmount(1);
bool LiveBar::isLive(true);
bool LiveBar::isImmortal( false );

/** */
LiveBar::LiveBar() :pDistStr( Property::get("HIGH_GAME_DIST") ), pBonusAmount( 0 ), timer(),
pImmortalTexture( 0 ), pImmortalOpacity( 0.0f ), pBonusIcon( SpriteManager::getInstance()->getSprite("HEARTH_ADD") )
{
  pDistanceRect.y = Property::getSetting( "HIGH_OFFSET_Y" );
  pDistanceRect.x = (App::getScreenWidth() / 2) -100;
  
}


/** Rysowanie paska zycia...cala gorna czesc ekranu */
void LiveBar::draw() {

	//@TODO optymalizacja

	// kiedy wlaczona niesmiertelnosc pokazuje sie obwodka dookola ektanu */
	if( pImmortalTexture == 0 ) {
		logger.warring("Get surface in GL fomrat");
		pImmortalTexture = Renderer::getSurfaceInGLFormat( Resource::getSurf("FILL") );
	}

    Rect tmp(0,0,pScreenWidth,pScreenHeight);

    tmp.a= pImmortalOpacity;

    pRendererPtr->draw( pImmortalTexture , tmp);

	pRendererPtr->draw( "LIVE_BAR_BACK", pScreenWidth*0.053, pScreenHeight*0.043, -1, -1 );

	if( pLiveAmount > 0.0f )
		pRendererPtr->draw( "LIVE_BAR_FILL", pScreenWidth*0.053, pScreenHeight*0.044, 200*pLiveAmount, -1 );

	pRendererPtr->draw( "LIVE_BAR",      pScreenWidth*0.016,  pScreenHeight*0.025, -1, -1 );

	// Wypisywanie informacji i przebytej drodze

    //@TODO znalsc wszystkie okreslenia czcionki poprzez liczbe magiczna i zastaoic stala
    pWriterPtr->draw( pDistanceRect, pDistStr + toString( pDistNum ) + " m" );


    Rect textRect( pScreenWidth - 140, 45 );
    pWriterPtr->draw( textRect, "x" , FontSize::SMALL );
    textRect.x += 30;
    textRect.y -= 10;
    pWriterPtr->draw( textRect, toString( pBonusAmount ), FontSize::BIG );

    //@TODO optymalizacja
    pBonusIcon.draw( pScreenWidth - 200, 30 );

}


/** */
void LiveBar::update(const float& dt ) {
	if( isImmortal ) return;

	float nextLiveAmount = 0.0f;

//@TODO jakos dziwnie dziala- przetestowac
  if( Player::isFly() )
	  nextLiveAmount = pLiveAmount - (dt*0.1);
  else
	  nextLiveAmount = pLiveAmount - (dt*0.25);

  pLiveAmount = nextLiveAmount;

  if( pImmortalOpacity > 0 )
	  pImmortalOpacity -= 0.02;

}

/** */
void LiveBar::colision( short type ) {
//@TODO usunac liczby magiczne

	if( type == COLISION_WITH_ENEMY ) {
		debug("Enemy colision");
		pLiveAmount -= 0.3;
	}
	else {
		 pLiveAmount -= 0.1;
	}
  
}

/** */
bool LiveBar::useBonus() {
	if( isImmortal == true ) return false;

	  if( pBonusAmount > 0 ){
		  --pBonusAmount;
		  isImmortal = true;
		  pImmortalOpacity = 1.0f;
		  timer = SDL_AddTimer( 4000, disableImmortal_callbackTimer, this );
		  pLiveAmount = 1.0f;
		  return true;
	  }
	  else {
		  return false;
	  }
}

/** Metoda statyczna wywolywana przez timer wtedy kiedy mija czas w ktorym gracz nie traci zycia */
Uint32 LiveBar::disableImmortal_callbackTimer(Uint32 interval, void* param) {
	isImmortal = false;
	return( 0 );
}

/** */
void LiveBar::increaseLive() {
    if( pLiveAmount <= 0.9 )
    pLiveAmount += 0.15;
}

/** */
void LiveBar::reset() {
  pLiveAmount = 1.0f;
  isLive = true;
  pBonusAmount = 0;
}
